workflow/query.py

```diff
@@ -76,6 +76,107 @@ def insert:source_has_input_format(r,p) =
     """
 
 
+def discover_reimport_sources(sources: List[Source], force_reimport: bool, force_reimport_not_chunk_partitioned: bool) -> str:
+    date_partitioned_src_cfg_csv = ""
+    for src in sources:
+        date_partitioned_src_cfg_csv += f"{src.to_chunk_partitioned_paths_csv()}\n"
+    return f"""
+        def force_reimport = {"true" if force_reimport else "false"}
+        def force_reimport_not_chunk_partitioned = {"true" if force_reimport_not_chunk_partitioned else "false"}
+    
+        def resource_config[:data] = \"\"\"{date_partitioned_src_cfg_csv}\"\"\"
+        def resource_config[:syntax, :header_row] = -1
+        def resource_config[:syntax, :header] = (1, :Relation); (2, :Path); (3, :ChunkPartitioned)
+        def resource_config[:schema, :Relation] = "string"
+        def resource_config[:schema, :Path] = "string"
+        def resource_config[:schema, :ChunkPartitioned] = "string"
+
+        def source_config_csv = load_csv[resource_config]
+
+        def chunk_partitioned_sources(rel, path, p_idx) {{
+            source_config_csv(:Path, i, path) and
+            source_config_csv(:Relation, i, rel) and
+            source_config_csv(:ChunkPartitioned, i, "True") and
+            p_idx = part_resource:parse_part_index[path]
+            from i
+        }}
+        
+        def simple_sources(rel, path) {{ 
+            source_config_csv(:Path, i, path) and
+            source_config_csv(:Relation, i, rel) and
+            not chunk_partitioned_sources(rel, path, _)
+            from i
+        }}
+        // TODO: add support for chunk partitioned sources which are not date partitioned
+        /*
+         * All simple sources are affected if they match with declared sources.
+         */
+        def potentially_affected_sources(rel, path) {{
+            source_declares_resource(rel, path) and
+            simple_sources(rel, path)
+        }}
+        /*
+         * All chunk partitioned sources for given date are affected in case new sources have at least one partition in this date.
+         */
```

Comment:
could some part of this code go installed? with the flags being bounds and populated here, but the rest of the logic installed?

Reply:
Moved